src/Agent.Sdk/Util/VssUtil.cs

```diff
@@ -21,6 +21,21 @@ public static class VssUtil
     {
         private static UtilKnobValueContext _knobContext = UtilKnobValueContext.Instance();
 
+        private const string _testUri = "https://microsoft.com/";
```

Comment:
In testing I've found that reaching this URL may take anywhere between 1 and 5 seconds, which produces a bit of a slowdown on startup of the agent, and also during configuration. Maybe we could use some local URL here, but then the results may vary depending on user's setup.
@anatolybolshakov is it worth the effort to investigate this?

Reply:
Could it make sense to just exclude OS for which we found that it's not supported?
There are other possible drawbacks of this approach:
- How it behaves with proxy enabled?
- Performance
-

Reply:
We discussed, I think we can just minimize the use of this check, e.g. check only if using AZP_AGENT_USE_LEGACY_HTTP knob and the current system is macOS or Linux. Excluding linux would not be quite right, because in most cases everything works correctly.